myapp/forms.py

```diff
@@ -0,0 +1,24 @@
+from django import forms
+from django.contrib.auth.models import User
+from django.contrib.auth.forms import UserCreationForm, PasswordChangeForm, AuthenticationForm
+
+
+class CustomRegistrationFrom(UserCreationForm):
+    username = forms.CharField(label='username', required=False, widget=forms.TextInput(attrs={'class': 'form-control', "placeholder": "Username", "rows": 1, "cols": 22}))
+    email = forms.EmailField(label='email', max_length=100, widget=forms.TextInput(attrs={'class': 'form-control', "placeholder": "Email address", "rows": 1, "cols": 22}))
+    password1 = forms.CharField(label='password1', min_length=4, max_length=30, widget=forms.PasswordInput(attrs={'class': 'form-control', "placeholder": "Create password", "rows": 1, "cols": 20}))
+    password2 = forms.CharField(label='password2', min_length=4, max_length=30, widget=forms.PasswordInput(attrs={'class': 'form-control', "placeholder": "Confirm password", "rows": 1, "cols": 20}))
+
+    class Meta:
+        model = User
+        fields = ['username', 'email', 'password1', 'password2']
```

Comment:
the fields which already exist in the model `User`  no need to redefine (unless you have custom field attributes), for example:
```
class UserForm(forms.ModelForm):
    class Meta:
        model = User
        fields = ('username', 'email', 'password')
        widgets = {'username': forms.TextInput(attrs={
            "placeholder": "Username"
        }),
            "email": forms.EmailInput(attrs={
                "placeholder": "Email"
            }),
            "password": forms.PasswordInput(attrs={'placeholder': 'Password'})

        }
```

Reply:
I am redefining them just to customize the widget CSS (purely for aesthetic reasons)